src/Collections/RepositoryCollection.php

```diff
@@ -67,6 +67,25 @@ public function canBeFilteredByRepository()
         return true;
     }
 
+    private $count = null;
+
+    public function count()
+    {
+        if ($this->count === null){
+            $this->prepareCollectionForExecution();
+            $this->count = $this->getRepository()->countRowsInCollection($this);            
+        }
+
+        return $this->count;
+    }
+
+    protected function invalidate()
+    {
+        parent::invalidate();
+
+        $this->count = null;
```

Comment:
To keep consistency with some of the rest of this code, should we perhaps have a `clearCount()` function?

Reply:
There is a difference just in that the clearXX() methods are there for an external user of the class to manipulate the configuration of the collection whereas the count is a private caching feature so I don't necessarily see a need to wrap this in a function call.